services/galley/src/Galley/Intra/Journal.hs

```diff
@@ -32,40 +33,35 @@ import qualified Galley.Aws as Aws
 -- Team journal operations to SQS are a no-op when the service
 -- is started without journaling arguments
 
-teamCreate :: TeamId -> UserId -> Galley ()
-teamCreate tid uid = do
-    mEnv <- view aEnv
-    for_ mEnv $ \e -> do
-        now <- nowInt
-        let event = TeamEvent TeamEvent'TEAM_CREATE (bytes tid) now (Just (evData 1 [uid]))
-        Aws.execute e (Aws.enqueue event)
+teamActivate :: TeamId -> [TeamMember] -> Galley ()
+teamActivate tid mems = journalEvent TeamEvent'TEAM_ACTIVATE tid (Just $ evData mems)
 
 teamUpdate :: TeamId -> [TeamMember] -> Galley ()
-teamUpdate tid mems = do
-    mEnv <- view aEnv
-    for_ mEnv $ \e -> do
-        now <- nowInt
-        let bUsers = view userId <$> filter (`hasPermission` SetBilling) mems
-        let eData = evData (fromIntegral $ length mems) bUsers
-        let event = TeamEvent TeamEvent'TEAM_UPDATE (bytes tid) now (Just eData)
-        Aws.execute e (Aws.enqueue event)
+teamUpdate tid mems = journalEvent TeamEvent'TEAM_UPDATE tid (Just $ evData mems)
 
 teamDelete :: TeamId -> Galley ()
-teamDelete tid = do
-    mEnv <- view aEnv
-    for_ mEnv $ \e -> do
-        now <- nowInt
-        let event = TeamEvent TeamEvent'TEAM_DELETE (bytes tid) now Nothing
-        Aws.execute e (Aws.enqueue event)
+teamDelete tid = journalEvent TeamEvent'TEAM_DELETE tid Nothing
+
+teamSuspend :: TeamId -> Galley ()
+teamSuspend tid = journalEvent TeamEvent'TEAM_SUSPEND tid Nothing
+
+journalEvent :: TeamEvent'EventType -> TeamId -> Maybe TeamEvent'EventData -> Galley ()
+journalEvent typ tid dat = view aEnv >>= \mEnv -> for_ mEnv $ \e -> do
+    now <- nowInt
```

Comment:
The timestamps sent on TEAM_ACTIVATE are currently not consistent:

- upon first creating then activating a team, T2, the timestamp at email activation (not creation) is used 
- upon journaling, T1, the timestamp of the team_creation (not activation) will be used
- in case a team is suspended then re-activated, T3, the time of re-activation is used. 

It might be simplest to use the team creation timestamp (e.g. `writetime(binding)`) everywhere?

Reply:
> It might be simplest to use the team creation timestamp (e.g. writetime(binding)) everywhere?

👍